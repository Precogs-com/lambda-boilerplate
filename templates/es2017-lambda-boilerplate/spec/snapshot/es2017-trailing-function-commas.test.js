const { transform } = require('./utils');

// Trailing commas in function expression signature

it('variable assignment with function primitive', () => {
    const code = transform(`
var f = function(
    a,
    b,
    c,
) { return a + b + c; }
`);
    expect(code).toMatchSnapshot();
});

it('named function', () => {
    const code = transform(`
function f(
    a,
    b,
    c,
) { return a + b + c; }
`);
    expect(code).toMatchSnapshot();
});

it('IIFE', () => {
    const code = transform(`
(function(
    a,
    b,
    c,
) { return a + b + c; })();
`);
    expect(code).toMatchSnapshot();
});

it('named IIFE', () => {
    const code = transform(`
(function f(
    a,
    b,
    c,
) { return a + b + c; })();
`);
    expect(code).toMatchSnapshot();
});

it('array primitive declaration with function primitive', () => {
    const code = transform(`
var arr = [function(
    a,
    b,
    c,
) { return a + b + c; }];
`);
    expect(code).toMatchSnapshot();
});

it('array primitive declaration with named function primitive', () => {
    const code = transform(`
var arr = [function f(
    a,
    b,
    c,
) { return a + b + c; }];
`);
    expect(code).toMatchSnapshot();
});

it('Array() constructor declaration with function primitive', () => {
    const code = transform(`
var arr = Array(function(
    a,
    b,
    c,
) { return a + b + c; });
`);
    expect(code).toMatchSnapshot();
});

it('Array() constructor declaration with named function primitive', () => {
    const code = transform(`
var arr = Array(function f(
    a,
    b,
    c,
) { return a + b + c; });
`);
    expect(code).toMatchSnapshot();
});

it('object primitive declaration with function primitive', () => {
    const code = transform(`
var arr = { x: function(
    a,
    b,
    c,
) { return a + b + c; }};
`);
    expect(code).toMatchSnapshot();
});

it('object primitive declaration with named function primitive', () => {
    const code = transform(`
var arr = { x : function f(
    a,
    b,
    c,
) { return a + b + c; }};
`);
    expect(code).toMatchSnapshot();
});

it('object primitive declaration with named function shorthand', () => {
    const code = transform(`
var arr = { f(
    a,
    b,
    c,
) { return a + b + c; }};
`);
    expect(code).toMatchSnapshot();
});

// Trailing commas in arrow function signature

it('variable assignment with arrow function primitive', () => {
    const code = transform(`
var f = (
    a,
    b,
    c,
) => { return a + b + c; }
`);
    expect(code).toMatchSnapshot();
});

it('arrow function IIFE', () => {
    const code = transform(`
((
    a,
    b,
    c,
) => { return a + b + c; })();
`);
    expect(code).toMatchSnapshot();
});

it('array primitive declaration with arrow function primitive', () => {
    const code = transform(`
var arr = [(
    a,
    b,
    c,
) => { return a + b + c; }];
`);
    expect(code).toMatchSnapshot();
});

it('Array() constructor declaration with function primitive', () => {
    const code = transform(`
var arr = Array((
    a,
    b,
    c,
) => { return a + b + c; });
`);
    expect(code).toMatchSnapshot();
});

it('object primitive declaration with arrow function primitive', () => {
    const code = transform(`
var arr = { x: (
    a,
    b,
    c,
) => { return a + b + c; }};
`);
    expect(code).toMatchSnapshot();
});

// Trailing commas in function call

it('function call expression', () => {
    const code = transform(`
function f(a, b) { return a + b; }
var x = f(
    1,
    2,
);
`);
    expect(code).toMatchSnapshot();
});

it('IIFE call', () => {
    const code = transform(`
(function f(a, b) { return a + b; })(
    1,
    2,
);
`);
    expect(code).toMatchSnapshot();
});
