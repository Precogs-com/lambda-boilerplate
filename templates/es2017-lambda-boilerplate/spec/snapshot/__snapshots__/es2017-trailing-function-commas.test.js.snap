// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Array() constructor declaration with function primitive 1`] = `
"\\"use strict\\";

var arr = Array(function (a, b, c) {
    return a + b + c;
});"
`;

exports[`Array() constructor declaration with function primitive 2`] = `
"\\"use strict\\";

var arr = Array((a, b, c) => {
    return a + b + c;
});"
`;

exports[`Array() constructor declaration with named function primitive 1`] = `
"\\"use strict\\";

var arr = Array(function f(a, b, c) {
    return a + b + c;
});"
`;

exports[`IIFE 1`] = `
"\\"use strict\\";

(function (a, b, c) {
    return a + b + c;
})();"
`;

exports[`IIFE call 1`] = `
"\\"use strict\\";

(function f(a, b) {
    return a + b;
})(1, 2);"
`;

exports[`array primitive declaration with arrow function primitive 1`] = `
"\\"use strict\\";

var arr = [(a, b, c) => {
    return a + b + c;
}];"
`;

exports[`array primitive declaration with function primitive 1`] = `
"\\"use strict\\";

var arr = [function (a, b, c) {
    return a + b + c;
}];"
`;

exports[`array primitive declaration with named function primitive 1`] = `
"\\"use strict\\";

var arr = [function f(a, b, c) {
    return a + b + c;
}];"
`;

exports[`arrow function IIFE 1`] = `
"\\"use strict\\";

((a, b, c) => {
    return a + b + c;
})();"
`;

exports[`function call expression 1`] = `
"\\"use strict\\";

function f(a, b) {
    return a + b;
}
var x = f(1, 2);"
`;

exports[`named IIFE 1`] = `
"\\"use strict\\";

(function f(a, b, c) {
    return a + b + c;
})();"
`;

exports[`named function 1`] = `
"\\"use strict\\";

function f(a, b, c) {
    return a + b + c;
}"
`;

exports[`object primitive declaration with arrow function primitive 1`] = `
"\\"use strict\\";

var arr = { x: (a, b, c) => {
        return a + b + c;
    } };"
`;

exports[`object primitive declaration with function primitive 1`] = `
"\\"use strict\\";

var arr = { x: function (a, b, c) {
        return a + b + c;
    } };"
`;

exports[`object primitive declaration with named function primitive 1`] = `
"\\"use strict\\";

var arr = { x: function f(a, b, c) {
        return a + b + c;
    } };"
`;

exports[`object primitive declaration with named function shorthand 1`] = `
"\\"use strict\\";

var arr = { f(a, b, c) {
        return a + b + c;
    } };"
`;

exports[`variable assignment with arrow function primitive 1`] = `
"\\"use strict\\";

var f = (a, b, c) => {
    return a + b + c;
};"
`;

exports[`variable assignment with function primitive 1`] = `
"\\"use strict\\";

var f = function (a, b, c) {
    return a + b + c;
};"
`;
